src/protocol/modulus_conversion/convert_shares.rs

```diff
@@ -144,23 +145,29 @@ where
     C: Context<F, Share = S>,
     S: ArithmeticSecretSharing<F>,
 {
-    let ctx = ctx.set_total_records(locally_converted_bits.len());
+    let ctx = ctx.set_total_records(num_records);
 
     let all_bits = (0..num_bits as usize).collect::<Vec<_>>();
-    try_join_all(all_bits.chunks(num_multi_bits as usize).map(|chunk| {
-        try_join_all(
-            zip(locally_converted_bits, repeat(ctx.clone()))
-                .enumerate()
-                .map(|(idx, (record, ctx))| async move {
-                    convert_bit_list(
-                        ctx.narrow(&ModulusConversion(chunk[0].try_into().unwrap())),
-                        &chunk.iter().map(|i| &record[*i]).collect::<Vec<_>>(),
-                        RecordId::from(idx),
-                    )
-                    .await
-                }),
+    try_join_all(
+        zip(
+            repeat(&locally_converted_bits),
+            all_bits.chunks(num_multi_bits as usize),
         )
-    }))
+        .map(|(locally_converted_bits, chunk)| {
+            try_join_all(
+                zip(*locally_converted_bits, repeat(ctx.clone()))
```

Comment:
Need help here. We basically want to iterate multiple times using the same iterator. How to do this? Error: cannot move out of `*locally_converted_bits` which is behind a shared reference